features/FEATURE_BLE/ble/gap/Gap.h

```diff
@@ -538,6 +538,12 @@ class Gap {
      */
     virtual uint16_t getMaxAdvertisingDataLength();
 
+    /** Return maximum advertising data length supported for connectable advertising.
+     *
+     * @return Maximum advertising data length supported for connectable advertising.
+     */
+    virtual uint16_t getMaxConnectableAdvertisingDataLength();
```

Comment:
I really don't think we should expose that information to the user code as it is non standard. 
Let's keep it internal for now and return error if the advertising payload set doesn't fit. We can mention limitation in the documentation. 

But until we sorted out why the Cordio stack makes a difference between connectable and non connectable data we should not create a new user facing API.